pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -440,23 +451,127 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
+	// TODO: in UPS operator version 0.x, we were using DCs and ImageStreams.
+	// TODO: in 1.0.0, we introduced this code block to migrate from old resources to new ones.
+	// TODO: in 2.0.0, we should get rid of this migration block, as well as unneeded permissions
+	// TODO: to access these old resources.
+	clientset, err := kubernetes.NewForConfig(r.config)
+
+	dcResourceExists, err := ApiVersionExists(clientset.DiscoveryClient, "apps.openshift.io/v1")
+	if err != nil {
+		reqLogger.Error(err, "Unable to check if a OpenShift's apps.openshift.io/v1 api version is available.")
+		return reconcile.Result{}, err
+	}
+
+	imageStreamResourceExists, err := ApiVersionExists(clientset.DiscoveryClient, "image.openshift.io/v1")
+	if err != nil {
+		reqLogger.Error(err, "Unable to check if a OpenShift's image.openshift.io/v1 api version is available.")
+		return reconcile.Result{}, err
+	}
+
+	if dcResourceExists {
+		//#region DELETE Postgres DeploymentConfig as we moved to Kube Deployments now
+		postgresDeploymentConfigObjectMeta := objectMeta(instance, "postgresql")
+		foundPostgresqlDeploymentConfig := &openshiftappsv1.DeploymentConfig{}
+		err = operatorClient.Get(context.TODO(), types.NamespacedName{Name: postgresDeploymentConfigObjectMeta.Name, Namespace: postgresDeploymentConfigObjectMeta.Namespace}, foundPostgresqlDeploymentConfig)
+		if err != nil && !errors.IsNotFound(err) {
+			// if there is another error than the DC not being found
+			reqLogger.Error(err, "Unable to check if a DeploymentConfig exists for Postgres.", "DeploymentConfig.Namespace", foundPostgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", foundPostgresqlDeploymentConfig.Name)
+			return reconcile.Result{}, err
+		} else if err == nil {
+			reqLogger.Info("Found a DeploymentConfig for Postgres. Deleting it.", "DeploymentConfig.Namespace", foundPostgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", foundPostgresqlDeploymentConfig.Name)
+			err = operatorClient.Delete(context.TODO(), foundPostgresqlDeploymentConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		//#endregion
+
+		//#region DELETE Unified Push Server DeploymentConfig as we moved to Kube Deployments now
+		upsDeploymentConfigObjectMeta := metav1.ObjectMeta{
+			Namespace: instance.Namespace,
+			Name:      instance.Name, // this is the name of the DeploymentConfig we were using
+		}
+		foundUpsDeploymentConfig := &openshiftappsv1.DeploymentConfig{}
+		err = operatorClient.Get(context.TODO(), types.NamespacedName{Name: upsDeploymentConfigObjectMeta.Name, Namespace: upsDeploymentConfigObjectMeta.Namespace}, foundUpsDeploymentConfig)
+		if err != nil && !errors.IsNotFound(err) {
+			// if there is another error than the DC not being found
+			reqLogger.Error(err, "Unable to check if a DeploymentConfig exists for Unified Push Server.", "DeploymentConfig.Namespace", foundUpsDeploymentConfig.Namespace, "DeploymentConfig.Name", foundUpsDeploymentConfig.Name)
```

Comment:
```suggestion
			reqLogger.Error(err, "Unable to check if a DeploymentConfig exists for UnifiedPush Server.", "DeploymentConfig.Namespace", foundUpsDeploymentConfig.Namespace, "DeploymentConfig.Name", foundUpsDeploymentConfig.Name)
```

This is not important, just highlighting how "UnifiedPush Server" is typically styled here: https://github.com/aerogear/aerogear-unifiedpush-server

Reply:
Fixed